### ISSUE TITLE
peak_local_max does not suppress close-by maxima

### ISSUE THREAD

**Issue by funkey** [scikit-image/scikit-image#1831]
When running peak_local_max on the following image (distance transform of binary image, isolines in same color):
![fp_distance](https://cloud.githubusercontent.com/assets/776708/11844702/73b943da-a410-11e5-904b-09972b55867f.png)
via

``` python
maxima = master_peak_local_max(distance, min_distance=10, threshold_abs=0, threshold_rel=0, indices=False)
```

I get the following result:
![fp_maxima](https://cloud.githubusercontent.com/assets/776708/11844708/792b8102-a410-11e5-8299-408c9897e2cc.png)

The highlighted pixels are indeed local maxima (not visible in distance image above), but closer than `2*min_distance+1` to each other.

Example image to reproduce: https://www.dropbox.com/s/2e7yfnuiirms36w/distance.tif?dl=0

**Comment by soupault** [comment]
@funkey hi!
Could you define the interval you presented in images above so it will be easier for us to debug? Please, do it for this example and also in https://github.com/scikit-image/scikit-image/issues/1832 .

**Comment by soupault** [comment]
Code to reproduce with region specified above:
```
import matplotlib
import matplotlib.pyplot as plt
import skimage
from skimage import io
from skimage.feature import peak_local_max

distance = io.imread('./distance.tif')
distance = distance[550:590, 620:690]  # Issue #1831

maxima = peak_local_max(distance, min_distance=10, threshold_abs=0, threshold_rel=0, indices=False)

fig, axes = plt.subplots(nrows=1, ncols=2)
axes[0].imshow(distance, interpolation='none',cmap='coolwarm')
axes[1].imshow(maxima, interpolation='none', cmap='gray')
plt.tight_layout()
plt.savefig('test_img.png')
plt.close()
```
![test_img_1831](https://cloud.githubusercontent.com/assets/1315589/12866515/7d6b7a98-cce1-11e5-8a55-7951f0020228.png)

**Comment by soupault** [comment]
with `exclude_border=False`:
![test_img_1831_exclude](https://cloud.githubusercontent.com/assets/1315589/12866558/c0d410fa-cce2-11e5-8469-445174fd324e.png)

**Comment by ahojnnes** [comment]
It is doing the right thing, imo. The three peaks in the middle have the same value.

**Comment by soupault** [comment]
The documentation says:
`If peaks are flat (i.e. multiple adjacent pixels have identical intensities), the coordinates of all such pixels are returned.`
So, I suppose, one would expect that function should return only one of these peaks.

**Comment by ahojnnes** [comment]
The documentation correctly describes what the function is doing. Look into `corner_peaks` if you only want one of the peaks.

**Comment by soupault** [comment]
I'd propose to change the description from
`Peaks are the local maxima in a region of 2 * min_distance + 1 (i.e. peaks are separated by at least min_distance).` to
`Peaks are the local maxima in a region of 2 * min_distance + 1 (i.e. peaks of different intensities are separated by at least min_distance).`.
This will close the issue, in my opinion.

**Comment by ahojnnes** [comment]
Good suggestion, do you want to go ahead and create a PR with a fix?

**Comment by jni** [comment]
@ahojnnes how is this correct? "multiple **-->adjacent<--** pixels have identical intensities" The above pixels are not adjacent, even if they have the same value.

**Comment by ahojnnes** [comment]
@jni That's why we created the PR, feel free to contribute to the discussion there.

**Comment by b52** [comment]
So is this an issue related to `scipy.ndimage.filters.maximum_filter` or the processing done in skimage? If the former is the case, one could look at different approaches like [efficient non-maximum suppression](http://www.vision.ee.ethz.ch/publications/papers/proceedings/eth_biwi_00446.pdf).

**Comment by soupault** [comment]
@b52 I think to the latter.
NMS is a very nice feature, by the way. If you find it useful, please, don't hesitate to raise a 'feature request`-issue.

**Comment by rfezzani** [comment]
Fixed by #3600, Closing ;-)
